db-acra-migration/python/migration.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2016, Cossack Labs Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# coding: utf-8
+import argparse
+import io
+
+from sqlalchemy import (Table, Column, Integer, MetaData, select, update, Text, func, asc,)
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.dialects import postgresql
+
+from common import get_engine, register_common_cli_params
+
+metadata = MetaData()
+user_table = Table(
+    'users', metadata,
+    Column('id', Integer, primary_key=True, nullable=False),
+    Column('phone_number', Text, nullable=False),
+    Column('ssn', Text, nullable=False),
+    Column('email', Text, nullable=False),
+    Column('firstname', Text, nullable=False),
+    Column('lastname', Text, nullable=False),
+    Column('age', Integer, nullable=False),
+)
+
+
+def generate_data(engine, csv_string):
+    session = sessionmaker(bind=engine)()
+
+    cursor = session.connection().connection.cursor()
+    cursor.copy_expert(
+        f'COPY users (id,phone_number,ssn,email,firstname,lastname,age) FROM STDIN WITH (FORMAT CSV, HEADER)',
+        io.StringIO(csv_string),
+    )
+    session.commit()
+    print("Data generated successfully!")
+
+
+def print_data(connection, table=user_table):
+    """fetch data from database and print to console"""
+    query = select(user_table).order_by(asc(user_table.c.id)).limit(10)
+    print("Fetch data by query {}\n",
+          query.compile(dialect=postgresql.dialect(), compile_kwargs={"literal_binds": True}))
+    result = connection.execute(query)
+    result = result.fetchall()
+
+    # print(len(result))
+    for row in result:
+        values = ['{:<3}'.format(row['id'])]
+        for col in row[1:]:
+            if isinstance(col, memoryview):
+                values.append(col.tobytes().decode("utf-8"))
+            else:
+                values.append(str(col))
+
+        print(' - '.join(values))
+
+def migrate_data(connection, table=user_table):
+    count = connection.execute(select([func.count()]).select_from(user_table)).fetchone()[0]
```

Comment:
same here, let's use raw sql queries because reading such SQL builder instruction may scary other developers or be harder to understand for DBAs.